### PR TITLE
Add form to create a new task

### PR DESCRIPTION
**Pull request by pbochis**
- add bower dependency for markdown support
- page to create task
- create component to render markdown text
- display instructions as markdown on task detail page

### CODE REVIEW

**Comment by lorenzleutgeb** [comment]
Y u no space after bullet point

**Comment by pbochis** [comment]
Do not merge yet.  Still pushing.

**Comment by pbochis** [comment]
Ready to merge

File: app/elements/task/task-detail.html [hunk 1]
```diff
@@ -42,7 +43,7 @@
 
 			<paper-material	class="card grid-row">
 				<h4>Instructions</h4>
-				<p>{{task.Instructions}}</p>
+				<render-markdown markdown="{{task.Instructions}}"></render-markdown>
```
Comment (lorenzleutgeb) [inline app/elements/task/task-detail.html]: This is wrong. Should be `task.Assignment.Instructions`

File: app/elements/task/task-type-selector.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<dom-module id="task-type-selector">
```
Comment (lorenzleutgeb) [inline app/elements/task/task-type-selector.html]: In the future, there won't be any task types. This is obsolete.

File: app/scripts/util.js [hunk 3]
```diff
@@ -83,5 +83,14 @@ var util = {
...
 		'Efficiency',
 		'Coding Speed'
+	],
+	languages: [
+		'java',
+		'py',
+		'c',
+		'js'
+	],
+	taskTypes: [
```
Comment (lorenzleutgeb) [inline app/scripts/util.js]: Obsolete

File: app/elements/task/task-form.html [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	<style>
+		iron-autogrow-textarea {
+			border-color: black;
+			border: solid;
+			background-color: white;
+			width: 70%;
+			margin: 10px 0;
+		}
+
+
```
Comment (lorenzleutgeb) [inline app/elements/task/task-form.html]: Unnecessary empty lines :unamused:

File: app/elements/task/task-form.html [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+			width: 90%;
+			padding: 5%;
+		}
+	</style>
+	<template>
+		<iron-ajax
+		  id="postTaskAjax"
+		  handle-as="json"
+		  on-response="onResponse"
+		  on-error="onError">
```
Comment (lorenzleutgeb) [inline app/elements/task/task-form.html]: Not indented with tabs :unamused:

**Comment by pbochis** [comment]
Fixes #60

File: app/elements/task/task-form.html [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+					this.assignment.Duration += this.hours * 36 * 1e11;
+				}
+				if (this.minutes > 0){
+					this.assignment.Duration += this.minutes * 6 * 1e10;
+				}
+				var skillWeights = {};
+				var i;
+				for (i = 0; i < this.skills.length; i++){
+					var skillValue = parseFloat(document.getElementById(this.skills[i]).value);
+					if (skillValue.toString() !== 'NaN' && skillValue >= 0 && skillValue <= 1){
```
Comment (lorenzleutgeb) [inline app/elements/task/task-form.html]: Why didn't you use [`isNaN`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/isNaN)?

File: app/elements/task/task-form.html [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+				this.task.Tasker = parseInt(this.task.Tasker);
+
+				console.log(this.task);
+				util.post(this.$.postTaskAjax, '/tasks', this.task);
+			},
+			onResponse: function(res){
+				page.redirect('/tasks/' + res.detail.response.Key);
+			},
+			onError: function(e){
+				console.log(e.detail.request);
```
Comment (lorenzleutgeb) [inline app/elements/task/task-form.html]: Why not call the global error handler thingy in util?

File: app/elements/task/task-form.html [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+						selectedLangs.push(this.languages[i]);
+					}
+				}
+				this.assignment.Endpoints = this.endpoints;
+				this.task.Assignment = this.assignment;
+				this.task.SkillWeights = skillWeights;
+				this.task.Languages = selectedLangs;
+				this.task.Tasker = parseInt(this.task.Tasker);
+
+				console.log(this.task);
```
Comment (lorenzleutgeb) [inline app/elements/task/task-form.html]: Why `console.log` this?

**Pull request merged**